PusherMainView/PusherMainView/PusherViewController.swift

```diff
@@ -105,3 +107,15 @@ extension PusherViewController: PusherInteractable {
         apnsAuthTokenRadioButton.state = state.authTokenRadioState
     }
 }
+
+// MARK: - NSTextFieldDelegate
+
+extension PusherViewController: NSTextFieldDelegate {
+    public func controlTextDidChange(_ notification: Notification) {
+        guard notification.object as? NSTextField == deviceTokenTextField else {
+            return
+        }
+        let deviceToken = deviceTokenTextField.stringValue
+        pusherStore.dispatch(actionType: .deviceToken(deviceToken))
```

Comment:
The store is being updated on each typed character. Is that ok?

Reply:
I think so.